### PR TITLE
Make sequencing abstract

### PR DESCRIPTION
**Pull request by perryibm**
Remove CLI manual use of pipe.js and use of parameters and use a new sequence kind and components field.

Use SequenceExec for sequences but with a NodejsExec-like implementation for now where the compoments meta-data is converted to parameters.

### CODE REVIEW

File: common/scala/src/whisk/core/entity/Exec.scala [hunk 1]
```diff
@@ -76,6 +76,10 @@ protected[core] case class JavaExec(jar: String, main: String) extends Exec(Exec
 
 protected[core] case class BlackBoxExec(image: String) extends Exec(Exec.BLACKBOX)
 
+protected[core] case class SequenceExec(code: String, components: Vector[String]) extends Exec(Exec.SEQUENCE) {
+    val image = "whisk/nodejsaction"
```
Comment (rabbah) [inline common/scala/src/whisk/core/entity/Exec.scala]: :/

File: common/scala/src/whisk/core/entity/Exec.scala [hunk 2]
```diff
@@ -76,6 +76,10 @@ protected[core] case class JavaExec(jar: String, main: String) extends Exec(Exec
 
 protected[core] case class BlackBoxExec(image: String) extends Exec(Exec.BLACKBOX)
 
+protected[core] case class SequenceExec(code: String, components: Vector[String]) extends Exec(Exec.SEQUENCE) {
```
Comment (rabbah) [inline common/scala/src/whisk/core/entity/Exec.scala]: `Vector[Namespace]` instead?

Comment (perryibm) [inline common/scala/src/whisk/core/entity/Exec.scala]: I was going to leave it as a string for this round as the underlying implementation is still with pipe.js.  Next round, Vector[String] becomes Vector[Namespace] and code field disappears.

File: common/scala/src/whisk/core/entity/Exec.scala (hunk 1 above)
Comment (perryibm) [inline common/scala/src/whisk/core/entity/Exec.scala]: Yes - as below.

File: common/scala/src/whisk/core/entity/Exec.scala [hunk 3]
```diff
@@ -133,6 +141,17 @@ protected[core] object Exec
                         case _                => None
                     }
                     NodeJSExec(code, init)
+                case Exec.SEQUENCE =>
```
Comment (rabbah) [inline common/scala/src/whisk/core/entity/Exec.scala]: thoughts for pedantic check for defined `code` and reject?

**Comment by rabbah** [comment]
Replaced with #709.

**Pull request closed without merging**